concepts/number-types/about.md

```diff
@@ -0,0 +1,160 @@
+
+```crystal
+1_i8.class
+# => Int8
+
+3.class
+# => Int32
+
+5_i128.class
+# => Int128
+```
+
+To convert between different integer types you can use the `to_i<bit>` method.
+The `to_i` method converts to the default integer type, which is `Int32`.
+
+```crystal
+1_i8.to_i16.class
+# => Int16
+
+2_i16.to_i.class
+# => Int32
+```
+
+## Unsigned integers types
+
+Table of all the [unsigned integer][uint] types in Crystal:
+
+| Type      | Size (bits) | Range                                                    |
+| --------- | ----------- | -------------------------------------------------------- |
+| `UInt8`   | 8           | 0 to 255                                                 |
+| `UInt16`  | 16          | 0 to 65_535                                              |
+| `UInt32`  | 32          | 0 to 4_294_967_295                                       |
+| `UInt64`  | 64          | 0 to 18_446_744_073_709_551_615                          |
+| `UInt128` | 128         | 0 to 340_282_366_920_938_463_463_374_607_431_768_211_455 |
+
+The unsigned integer types can only be positive.
+Otherwise are the unsigned integer types the same as the signed integer types.
+
+The reason for having unsigned integer types is that they can be used to store values that are always positive.
+This can be useful when you want to for example only work with indexes in an array.
+
+The default unsigned integer type is `UInt32`.
+To declare an unsigned integer with a specific type you can use the type name as a suffix, by adding `<number>_u<bit>`
+
+```crystal
+1_u8.class
+# => UInt8
+
+5_u128.class
+# => UInt128
+```
+
+To convert between different unsigned integer types you can use the `to_u<bit>` method.
+The `to_u` method converts to the default unsigned integer type, which is `UInt32`.
+
+```crystal
+1_u8.to_u16.class
+# => UInt16
+
+2_u16.to_u.class
+# => UInt32
+```
+
+## Floating point types
+
+Table of all the floating point types in Crystal:
+
+| Type      | Size (bits) | Range                |
+| --------- | ----------- | -------------------- |
+| `Float32` | 32          | 1.2E-38 to 3.4E+38   |
+| `Float64` | 64          | 2.3E-308 to 1.7E+308 |
+
+There are two different [floating point types][float], one is more precise than the other.
+
+Float32 is lighter compared to Float64, but has less precision.
+Float64 is heavier compared to Float32, but has more precision.
+
+The default floating point type is `Float64`.
+
+```crystal
+1.0_f32.class
+# => Float32
+
+2.0_f64.class
+# => Float64
+```
+
+To convert between different floating point types you can use the `to_f<bit>` method.
+The `to_f` method converts to the default floating point type, which is `Float64`.
+
+```crystal
+1.0_f64.to_f32.class
+# => Float32
+
+2.0_f64.to_f.class
+# => Float64
+```
+
+## Type after an operation
+
+When you do an operation between two numbers, the result will be of the type of the number with the highest precision.
```

Comment:
```suggestion
When you execute an operation between two numbers, the result will be of the type of the number with the highest precision.
```

(generally "do a" is something you want to avoid in english)